contents/performance/PerformanceProfiling.md

```diff
@@ -0,0 +1,137 @@
+# Performance Profiling
+
+Author: [Ong Heng Le](https://github.com/initialshl)
+
+## Overview
+
+A performance profiler is a tool which collects data (such as function timings) about 
+your program to identify the areas with performance issues, commonly used in code 
+optimization. There are two common types of performance profiling methods.
+
+* **CPU sampling**: Collects *samples* at fixed intervals, which provides an overview of your program's performance
+* **Instrumentation**: Collects detailed *elapsed times* for your program's function calls
+
+Learn [which profiling method to use for your program](https://blogs.msdn.microsoft.com/ejarvi/2005/04/07/the-choice-between-sampling-and-instrumentation/) 
+when choosing between CPU sampling and instrumentation.
+
+### Resources used:
+
+1. What is a software profiler?: [Profiling Overview](https://msdn.microsoft.com/en-us/library/bb384493(v=vs.110).aspx)
+1. Common performance profiling methods: [Understanding Performance Collection Methods](https://msdn.microsoft.com/en-us/library/dd264994.aspx)
```

Comment:
So as not to break the flow of the document, you can put the 'Resources' section at the end. Perhaps no need to say `used` and `recommended`, just `Resources` would do. When you describe the resource, you can mention how you used it in this doc and/or what readers can get more from that resource.